### ISSUE TITLE
Rebalance resurrected node

### ISSUE THREAD

**Issue by sfcgeorge** [flynn/flynn#2101]
Firstly Flynn is looking great, I'm so excited about it. I've got a 3 node cluster working on 1€ ArubaCloud VPSes and it seems to work.

I tried `poweroff` one node, the two remaining nodes did take over the core tasks, but a web app running on the offed node wasn't resurrected... until I refreshed the dashboard. That's really weird. I don't know if it was just coincidental timing, but I'd left it at least 5 minutes. Perhaps an indicator in the dashboard or command line of the state of the cluster and whether it has finished whatever it is it does.

So then I brought back online the offed node, and sure enough Flynn found it and got `discoverd` and `flannel` running. `flynn-server fix` says everything is fine. But none of the load has been rebalanced to the newly available node. More worry is `postgres`—now 2 instances are still running on the same machine so if that machine breaks there will be dataloss I believe. Automatic or manual rebalance would be nice then :)

**Comment by sfcgeorge** [comment]
Running `flynn-host stop xyz` manually on processes residing on overloaded nodes does cause them to be restarted on the newly resurrected node that has most free memory, so that's good. An automatic rebalance option would still be nice, but I guess this works. It's hard to know what should be killed though? It would also be useful if Flynn had a built in monitoring system with email alerts or something, and a "System Load" tab on the dashboard maybe.

I killed a `flannel` process and that took the whole node down. Is that intended? I guess that must be the one that handles cluster membership, but is there a way it could be made to auto-restart like if it crashes? This seems like a weak spot.

With just 2 nodes remaining (after killing a flannel) some processes seem to keep crashing. Older running ones remain, but the top handful keep disappearing and then being started again, only sticking around for a minute or so. All nodes have 1GB RAM as recommended and `free -m` seems to show there's space, so I don't know if this is a low memory issue or an actual Flynn bug. I don't know if they're important either as the dashboard and a demo app keep running fine.

```
flynn2-a51a71f0-1365-48ae-8674-6da7ebe84391  running  About a minute ago  blobstore       web
flynn2-10594a5e-3383-475f-b875-d478fe91dccf  running  About a minute ago  controller      worker
flynn2-585c30ff-15aa-4dce-9b1c-6a8803a2c927  running  About a minute ago  router          app
flynn2-e62a16c7-c2f8-4576-99b6-b9a1acc67b71  running  About a minute ago  postgres        web
```

**Comment by sfcgeorge** [comment]
Bringing the 3rd node back with `start flynn-host` causes the cluster to stabilise and processes to stop endlessly restarting. 

Therefore 1GB of RAM per node in a 3 node cluster is enough when stable, but not enough when one node goes down, so 2GB would be a better recommendation. 2GB is probably needed for single node too as services aren't spread out. Perhaps the docs should be updated?

**Comment by josephglanville** [comment]
That does sound reasonable, but were the services restarting because of lack of memory?
You should be able to see the logs if the crashed processes with `flynn-host log <job_id>`. You can get said job IDs from `flynn-host ps -a` which will show completed and crashed jobs as well as running jobs.

**Comment by josephglanville** [comment]
Also it would be great if you still have this cluster running to run `flynn-host collect-debug-info` on each machine and post the gist links. This would help tremendously in diagnosing what is going on. :)

**Comment by sfcgeorge** [comment]
Well `free -m`, when ignoring cache stuff, said around 400MB RAM available and lots of swap, so perhaps not lack of RAM unless Flynn is very sensitive.

Ok I ran those. `flynn3` is the one I killed `flannel` on about 2 hours ago, and it looks like `flynn2` tried to take over but processes kept crashing. The `sfcgeorge` process is my small Rails website used as a demo process, no database. I rand logs on a few of the crashed ones, lots of timeouts it seems? https://gist.github.com/sfcgeorge/846a4adeb23659499f5c

flynn1 logs: https://gist.github.com/7842a61e533ab3f9cd43
flynn2 logs: https://gist.github.com/e74c9026072a83a6ed51 
flynn3 logs: https://gist.github.com/a96350235fb324c051f0

It seems stable for the last hour, this is an impressive piece of kit :D thanks for everyone's hard work.
